clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -504,8 +504,8 @@ static Expected<StringRef> convertSPIRVToIR(StringRef Filename,
     return SPIRVToIRWrapperPath.takeError();
 
   // Create a new file to write the converted file to.
-  auto TempFileOrErr =
-      createOutputFile(sys::path::filename(ExecutableName), "bc");
+  auto TempFileOrErr = createOutputFile(
```

Comment:
I expect more sites to crop up in the future. We should try to create output files in a similar fashion. One quick pointer. In the driver code, I see that we generate unique names for tmp files using this:

std::error_code EC = llvm::sys::fs::getPotentiallyUniqueFileName(Twine(BaseName) + Twine("-%%%%%%.") + Ext, Path);

Using this 'driver' way, we can just have a single function to generate such filenames and keep calling that everywhere, instead of thinking about meaningful names. I am not sure if we need meaningful names here.

I am on the fence though. @sarnex and @jasonlizhengjian, WDYT?

Thanks